tests/ecosystem/upgrade/test_upgrade_ocp.py

```diff
@@ -0,0 +1,164 @@
+import logging
+import json
+
+from ocs_ci.framework import Config
+from ocs_ci.ocs import ocp
+from ocs_ci.utility.utils import TimeoutSampler
+from ocs_ci.framework.testlib import ManageTest
+from ocs_ci.ocs.cluster import CephCluster
+from ocs_ci.ocs.exceptions import CephHealthException
+
+logger = logging.getLogger(__name__)
+
+# TODO: add image type validation (ga to ga , nightly to nightly, newer than current etc.)
+
+
+class TestUpgradeOCP(ManageTest):
+    """
+    1. check cluster health
+    2. check OCP version
+    3. perform OCP upgrade
+    4. check all OCP ClusterOperators
+    5. check OCP version
+    5. monitor cluster health
+    """
+    ocp_o = ocp.OCP()
+
+    def test_upgrade_ocp(self):
+        """
+
+        Tests OCS stability when upgrading OCP
+
+        """
+        config = Config()
+        ceph_cluster = CephCluster()
+        ceph_cluster.enable_health_monitor()
+
+        target_image = config.UPGRADE['ocp_upgrade_version']
+        image_path = config.UPGRADE['ocp_upgrade_path']
+        self.cluster_operators = self.get_all_cluster_operators()
+        logger.info(f" oc version: {self.get_current_oc_version()}")
+        self.get_all_cluster_operators()
+        upgrade_operator_timeout = TimeoutSampler(
```

Comment:
usage of TimeoutSampler will not work like this. See https://github.com/red-hat-storage/ocs-ci/blob/f818d7b9a42cfac5045885cf8741b494b0310577/ocs_ci/ocs/node.py#L88 as a reference of how to use it